### ISSUE TITLE
Is there preprocessed data available?

### ISSUE THREAD

**Issue by YHDASHEN** [Tandon-A/emotic#9]
Hi,

I'm sorry to bother again. I don't know if it's possible to have access to the preprocessed data. I guess there are something wrong with my running. I always get an mAP of 0.21.

Best regards,
Hui

**Comment by Tandon-A** [comment]
Hello Hui, 

Have you tried the techniques I mentioned in the other issue? 
 
Initially, when I was passing the gradient through all the layers of the feature extractor models for both body and context stream, I also achieved a similar mAP as yours. Freezing the feature extractors helped improve the performance in my case. Do check if you are using pre-trained feature extractors. 

I don't have the preprocessed data available now. 
The script for preprocessing the data is the same. You can try executing the mat2py script in debug mode and take a look if the context and body images are being extracted properly.  

One more thing which you can try is increasing the body images size. You would need to change the size in [L190](https://github.com/Tandon-A/emotic/blob/master/mat2py.py#L190) in mat2py file. 

Best Regards, 
Abhishek

**Comment by YHDASHEN** [comment]
Hi Abhishek,

thank you for your reply. I assume it's not the issue of preprocessed data. But I'm not so clear about what you said: "Freezing the feature extractors helped improve the performance in my case." Could you please explain in detail? How to do it? Thank you in advance.

Best regards,
Hui

**Comment by Tandon-A** [comment]
Hello,

I was mentioning about the below part for freezing the extractors: 
```python
for param in model_context.parameters():
  param.requires_grad = False
for param in model_body.parameters():
  param.requires_grad = False
```

If you have directly followed the notebook then you should get above 25 mAP. 

I am not sure where the error is coming from, but you should try the below points: 

1.  Run the mat2py file in debug mode and check the context and body image, if they are being saved alright.
2.  If the data is correct then run an overfitting test. Try training the model on a very small subset of the train data and test on the same train data. You should see a high mAP in that case, as the model should memorize everything in the small training set used. You can use about 100-200 images and check.
3. If the overfitting test from step 2 works: Increase the training epochs (on the full dataset) and save the model with the lowest validation loss. Try testing with this best save model. You can increase the epochs to 100 and then apply an early stop of about 12 epochs. You can also change the StepLR scheduler to a ReduceLROnPlateau scheduler. 
If the overfitting test from step 2 doesn't work: The issue could be in your data. Maybe the npy file is being saved wrong. This step would require further debugging. Maybe you could try without npy file using the [emotic_csvdataset](https://github.com/Tandon-A/emotic/blob/master/emotic_dataset.py#L34)

Do let me know if I missed something or if the code needs some update. 

Best Regards, 
Abhishek

**Comment by YHDASHEN** [comment]
Hi Abhishek,

thank you so much. I missed that part. In my code they are set to true. Now I will train it again.

best regards,
Hui
